src/lib.rs

```diff
@@ -811,6 +862,41 @@ pub trait Perform {
     /// subsequent characters were ignored.
     fn esc_dispatch(&mut self, _intermediates: &[u8], _ignore: bool, _byte: u8) {}
 
+    /// Invoked when the beginning of a new SOS (Start of String) sequence is
+    /// encountered.
+    fn sos_start(&mut self) {}
+
+    /// Invoked when the beginning of a new APC (Application Program Command)
+    /// sequence is encountered.
+    fn apc_start(&mut self) {}
+
+    /// Invoked when the beginning of a new PM (Privacy Message) sequence is
+    /// encountered.
+    fn pm_start(&mut self) {}
+
+    /// Invoked for every valid character (0x20-0xFF) in a SOS (Start of String)
+    /// sequence.
```

Comment:
```suggestion
    /// Invoked for every byte (0x20-0xFF) in a SOS (Start of String) sequence.
```

Same comment applies for the other functions. Calling these "valid characters" could be misleading to consumers.